rollup-http/rollup-http-server/src/rollup/mod.rs

```diff
@@ -14,19 +14,57 @@
 // limitations under the License.
 //
 
-pub const ROLLUP_DEVICE_NAME: &str = "/dev/rollup";
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
 
-use serde::{Deserialize, Serialize};
 use std::io::ErrorKind;
-use std::os::unix::prelude::RawFd;
 
-mod bindings;
-pub use bindings::CARTESI_ROLLUP_ADDRESS_SIZE;
-pub use bindings::CARTESI_ROLLUP_ADVANCE_STATE;
-pub use bindings::CARTESI_ROLLUP_INSPECT_STATE;
+use libc::c_void;
+use serde::{Deserialize, Serialize};
+use validator::Validate;
+use regex::Regex;
+use lazy_static::lazy_static;
+
+include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
+#[derive(Clone)]
+pub struct RollupFd(*mut cmt_rollup_t);
+
+impl RollupFd {
```

Comment:
This name is misleading `fd` implies this is a file descriptor, but that is not true. This structure has other fields as well.